tools/cli/internal/cli/changelog/create.go

```diff
@@ -45,19 +49,59 @@ func (o *Opts) Run() error {
 		return err
 	}
 
-	hiddenEntries := changelog.NewNotHiddenEntries(entries)
+	notHiddenEntries := changelog.NewNotHiddenEntries(entries)
+	versionedEntries, err := changelog.NewEntriesBetweenRevisionVersions(o.revisionPath)
+	if err != nil {
+		return err
+	}
 
 	if o.dryRun {
 		log.Printf("Detected dry-run mode. No changes will be saved.\n")
 		return nil
 	}
 
-	err = changelog.SaveChangelog(o.newOutputFilePath(changelogFileName), entries, o.fs)
-	if err != nil {
+	if errSaveFile := openapi.SaveToFile(o.newOutputFilePath(changelogFileName), "", notHiddenEntries, o.fs); errSaveFile != nil {
+		return errSaveFile
+	}
+
+	if err := o.fs.MkdirAll(fmt.Sprintf("%s/%s", o.outputPath, changelogAllFolderName), 0o755); err != nil {
+		return err
+	}
+
+	if errSaveFile := openapi.SaveToFile(
+		o.newOutputFilePath(fmt.Sprintf("%s/%s", changelogAllFolderName, changelogAllFileName)), "", entries, o.fs); errSaveFile != nil {
+		return errSaveFile
+	}
+
+	if err := o.fs.MkdirAll(fmt.Sprintf("%s/%s", o.outputPath, versionChangelogFolderName), 0o755); err != nil {
+		return err
+	}
+
+	for _, entry := range versionedEntries {
```

Comment:
[q] not sure I remember why we need the versioned entries to bse saved, do we need them?

Reply:
my understanding is that the versioned entries (the `fromVersion_toVersion.json` files) are the ones shown to the user when comparing two versions

Reply:
hmmm, I don't remember, but I think that docs use changelog.json as well, but ok to keep them here until we find out